hash_practice/exercises.py

```diff
@@ -2,18 +2,36 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)

```

Comment:
As we possibly add an entry to `anagram_dict` for each word in `strings` (in the case no words are anagrams of each other), the space complexity here is O(n).